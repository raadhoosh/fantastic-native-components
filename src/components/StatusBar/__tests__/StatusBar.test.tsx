import StatusBar from "..";
import Enzyme, { shallow, ShallowWrapper } from "enzyme";
import React from "react";
import { Text } from "react-native";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter() });

describe("StatusBar", () => {
    describe("rendering", () => {
        let wrapper: ShallowWrapper;
        beforeEach(() => {
            wrapper = shallow(<StatusBar backgroundColor={"blue"} barStyle="light-content" />);
        });
        it("renders", () => {
            expect(wrapper.exists()).toBe(true);
        });
        it("toMatchSnapshot", () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});