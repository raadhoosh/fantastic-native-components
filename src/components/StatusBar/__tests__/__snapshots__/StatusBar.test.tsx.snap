// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBar rendering toMatchSnapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    backgroundColor="blue"
    barStyle="light-content"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <StatusBar
        animated={false}
        backgroundColor="blue"
        barStyle="light-content"
        showHideTransition="fade"
        translucent={true}
      />,
      "style": Array [
        Object {
          "height": 20,
        },
        Object {
          "backgroundColor": "blue",
        },
      ],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animated": false,
        "backgroundColor": "blue",
        "barStyle": "light-content",
        "showHideTransition": "fade",
        "translucent": true,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <StatusBar
          animated={false}
          backgroundColor="blue"
          barStyle="light-content"
          showHideTransition="fade"
          translucent={true}
        />,
        "style": Array [
          Object {
            "height": 20,
          },
          Object {
            "backgroundColor": "blue",
          },
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animated": false,
          "backgroundColor": "blue",
          "barStyle": "light-content",
          "showHideTransition": "fade",
          "translucent": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
